### ISSUE TITLE
PGP signing

### ISSUE THREAD

**Issue by InnovativeInventor** [pjrule/federal-election-data#1]
Great idea!
I think with all the paranoia regarding tampering with elections, maybe commits and the data should be pgp signed by as many people as possible.

**Comment by pjrule** [comment]
Yeah, I definitely like the idea of proving that the results haven't been tampered with. There's less of a risk here because this is data is historical and (relatively) easily verifiable against at least one government website, but it's still a valid concern. Do you know of an easy way to implement this via PGP signing or something else? One way to verify consistency is to just have multiple people run the same scripts on hotlinked official government sources and get the same result, but OCR errors sometimes necessitate manually adjusting the output. We could explicitly publish a small set of agreed-upon transforms/adjustments for each file (primarily for the pre-1996 data), but that adds another layer of complication...

BTW, would you be interested in contributing to this project in general? It's a large and somewhat tedious undertaking, but I think it fills a hole in the current publicly available data.

**Comment by InnovativeInventor** [comment]
I was thinking of having multiple people running the scripts and attaching their signatures when they get the same results.  Maybe we could make a signatures folder with the signatures of everyone who has independently verified the data? Also, with the manually adjusted output, I like your idea of having an agreed list of adjustments. And, all commits can easily be signed. [Here's](https://help.github.com/articles/signing-commits-using-gpg/) a bit more on that.

I'm happy to help out on this project. This will be very useful for researchers or other people to analyze. I'll be busy for sometime, but I will try to find free time for this.

**Comment by pjrule** [comment]
Yeah, I did some research on Git GPG signing after you mentioned it. At the most basic level, it looks pretty simple. I'm all for it!

**Comment by InnovativeInventor** [comment]
Great! The hardest part about GPG is verifying keys. Just add them to your Github account and Github will mark all your signed commits as trusted. I'll then see your GPG fingerprint and import it from a public key server. We could also use Keybase to verify keys (thats more complex and more secure) or even exchange them in person, but that's too much of a hassle I think.
